--- misc/build/curl-7.72.0/winbuild/MakefileBuild.vc	2020-08-17 00:28:40.000000000 +0200
+++ misc/build/curl-7.72.0/winbuild/MakefileBuild.vc	2022-04-23 19:57:39.594763000 +0200
@@ -115,7 +115,7 @@
 !IFNDEF WITH_DEVEL
 WITH_DEVEL   = ../../deps
 !ENDIF
-DEVEL_INCLUDE= $(WITH_DEVEL)/include
+DEVEL_INCLUDE= $(WITH_DEVEL)/inc/external
 DEVEL_LIB    = $(WITH_DEVEL)/lib
 
 !IF EXISTS("$(DEVEL_INCLUDE)")
@@ -233,7 +233,7 @@
 ZLIB_LIB_DIR = $(ZLIB_PATH)\lib
 ZLIB_LFLAGS  = $(ZLIB_LFLAGS) "/LIBPATH:$(ZLIB_LIB_DIR)"
 !ELSE
-ZLIB_INC_DIR = $(DEVEL_INCLUDE)
+ZLIB_INC_DIR = $(DEVEL_INCLUDE)\zlib
 ZLIB_LIB_DIR = $(DEVEL_LIB)
 !ENDIF
 
@@ -568,6 +568,7 @@
 	@if not exist "$(LIB_DIROBJ)\vtls" mkdir $(LIB_DIROBJ)\vtls
 	@if not exist "$(LIB_DIROBJ)\vssh" mkdir $(LIB_DIROBJ)\vssh
 	@if not exist "$(LIB_DIROBJ)\vquic" mkdir $(LIB_DIROBJ)\vquic
+	@if not exist "$(LIB_DIROBJ)\vssh" mkdir $(LIB_DIROBJ)\vssh
 
 $(CURL_DIROBJ):
 	@if not exist "$(CURL_DIROBJ)" mkdir $(CURL_DIROBJ)
@@ -590,6 +591,9 @@
 	$(CURL_CC) $(CFLAGS) /Fo"$@"  $<
 
 {$(LIBCURL_SRC_DIR)\vquic\}.c{$(LIB_DIROBJ)\vquic\}.obj:
+	$(CURL_CC) $(CFLAGS) /Fo"$@"  $<
+
+{$(LIBCURL_SRC_DIR)\vssh\}.c{$(LIB_DIROBJ)\vssh\}.obj:
 	$(CURL_CC) $(CFLAGS) /Fo"$@"  $<
 
 $(LIB_DIROBJ)\libcurl.res: $(LIBCURL_SRC_DIR)\libcurl.rc
