diff -Nur misc/build/icu/source/common/common.vcproj misc/build/icu/source/common/common.vcproj
--- misc/build/icu/source/common/common.vcproj	2025-01-01 05:17:57.155933000 +0200
+++ misc/build/icu/source/common/common.vcproj	2025-01-01 05:19:04.152887000 +0200
@@ -203,7 +203,7 @@
 		</Configuration>
 		<Configuration
 			Name="Release|x64"
-			OutputDirectory=".\x64\Release"
+			OutputDirectory=".\..\..\lib"
 			IntermediateDirectory=".\x64\Release"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
@@ -295,7 +295,7 @@
 		</Configuration>
 		<Configuration
 			Name="Debug|x64"
-			OutputDirectory=".\x64\Debug"
+			OutputDirectory=".\..\..\lib"
 			IntermediateDirectory=".\x64\Debug"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
diff -Nur misc/build/icu/source/i18n/i18n.vcproj misc/build/icu/source/i18n/i18n.vcproj
--- misc/build/icu/source/i18n/i18n.vcproj	2025-01-01 05:17:51.273387000 +0200
+++ misc/build/icu/source/i18n/i18n.vcproj	2025-01-01 05:24:12.394293000 +0200
@@ -209,7 +209,7 @@
 		</Configuration>
 		<Configuration
 			Name="Release|x64"
-			OutputDirectory=".\x64\Release"
+			OutputDirectory=".\..\..\lib"
 			IntermediateDirectory=".\x64\Release"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
@@ -301,7 +301,7 @@
 		</Configuration>
 		<Configuration
 			Name="Debug|x64"
-			OutputDirectory=".\x64\Debug"
+			OutputDirectory=".\..\..\lib"
 			IntermediateDirectory=".\x64\Debug"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
diff -Nur misc/build/icu/source/io/io.vcproj misc/build/icu/source/io/io.vcproj
--- misc/build/icu/source/io/io.vcproj	2025-01-01 05:18:06.743249000 +0200
+++ misc/build/icu/source/io/io.vcproj	2025-01-01 05:20:39.777087000 +0200
@@ -209,7 +209,7 @@
 		</Configuration>
 		<Configuration
 			Name="Release|x64"
-			OutputDirectory=".\x64\Release"
+			OutputDirectory=".\..\..\lib"
 			IntermediateDirectory=".\x64\Release"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
@@ -304,7 +304,7 @@
 		</Configuration>
 		<Configuration
 			Name="Debug|x64"
-			OutputDirectory=".\x64\Debug"
+			OutputDirectory=".\..\..\lib"
 			IntermediateDirectory=".\x64\Debug"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
diff -Nur misc/build/icu/source/layout/layout.vcproj misc/build/icu/source/layout/layout.vcproj
--- misc/build/icu/source/layout/layout.vcproj	2025-01-01 05:18:01.898212000 +0200
+++ misc/build/icu/source/layout/layout.vcproj	2025-01-01 05:21:22.121321000 +0200
@@ -209,7 +209,7 @@
 		</Configuration>
 		<Configuration
 			Name="Debug|x64"
-			OutputDirectory=".\x64\Debug"
+			OutputDirectory=".\..\..\lib"
 			IntermediateDirectory=".\x64\Debug"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
@@ -308,7 +308,7 @@
 		</Configuration>
 		<Configuration
 			Name="Release|x64"
-			OutputDirectory=".\x64\Release"
+			OutputDirectory=".\..\..\lib"
 			IntermediateDirectory=".\x64\Release"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
diff -Nur misc/build/icu/source/layoutex/layoutex.vcproj misc/build/icu/source/layoutex/layoutex.vcproj
--- misc/build/icu/source/layoutex/layoutex.vcproj	2025-01-01 05:17:53.481075000 +0200
+++ misc/build/icu/source/layoutex/layoutex.vcproj	2025-01-01 05:23:34.897976000 +0200
@@ -208,7 +208,7 @@
 		</Configuration>
 		<Configuration
 			Name="Release|x64"
-			OutputDirectory=".\x64\Release"
+			OutputDirectory=".\..\..\lib"
 			IntermediateDirectory=".\x64\Release"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
@@ -302,7 +302,7 @@
 		</Configuration>
 		<Configuration
 			Name="Debug|x64"
-			OutputDirectory=".\x64\Debug"
+			OutputDirectory=".\..\..\lib"
 			IntermediateDirectory=".\x64\Debug"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
diff -Nur misc/build/icu/source/tools/ctestfw/ctestfw.vcproj misc/build/icu/source/tools/ctestfw/ctestfw.vcproj
--- misc/build/icu/source/tools/ctestfw/ctestfw.vcproj	2025-01-01 05:18:03.176621000 +0200
+++ misc/build/icu/source/tools/ctestfw/ctestfw.vcproj	2025-01-01 05:22:55.393701000 +0200
@@ -207,7 +207,7 @@
 		</Configuration>
 		<Configuration
 			Name="Release|x64"
-			OutputDirectory=".\x64\Release"
+			OutputDirectory=".\..\..\..\lib"
 			IntermediateDirectory=".\x64\Release"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
@@ -302,7 +302,7 @@
 		</Configuration>
 		<Configuration
 			Name="Debug|x64"
-			OutputDirectory=".\x64\Debug"
+			OutputDirectory=".\..\..\..\lib"
 			IntermediateDirectory=".\x64\Debug"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
diff -Nur misc/build/icu/source/tools/toolutil/toolutil.vcproj misc/build/icu/source/tools/toolutil/toolutil.vcproj
--- misc/build/icu/source/tools/toolutil/toolutil.vcproj	2025-01-01 05:18:03.684971000 +0200
+++ misc/build/icu/source/tools/toolutil/toolutil.vcproj	2025-01-01 05:22:13.713520000 +0200
@@ -210,7 +210,7 @@
 		</Configuration>
 		<Configuration
 			Name="Release|x64"
-			OutputDirectory=".\x64\Release"
+			OutputDirectory=".\..\..\..\lib"
 			IntermediateDirectory=".\x64\Release"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
@@ -307,7 +307,7 @@
 		</Configuration>
 		<Configuration
 			Name="Debug|x64"
-			OutputDirectory=".\x64\Debug"
+			OutputDirectory=".\..\..\..\lib"
 			IntermediateDirectory=".\x64\Debug"
 			ConfigurationType="2"
 			InheritedPropertySheets="$(VCInstallDir)VCProjectDefaults\UpgradeFromVC71.vsprops"
