--- misc/boost_1_64_0/boost/math/tools/config.hpp.orig	2016-09-21 14:33:24 UTC
+++ misc//build/boost_1_64_0/boost/math/tools/config.hpp
@@ -28,7 +28,7 @@
 
 #include <boost/math/tools/user.hpp>
 
-#if (defined(__CYGWIN__) || defined(__FreeBSD__) || defined(__NetBSD__) \
+#if (defined(__CYGWIN__) || defined(__NetBSD__) \
    || (defined(__hppa) && !defined(__OpenBSD__)) || (defined(__NO_LONG_DOUBLE_MATH) && (DBL_MANT_DIG != LDBL_MANT_DIG))) \
    && !defined(BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS)
 #  define BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS
--- misc/boost_1_64_0/boost/serialization/array.hpp.orig	2017-04-17 02:22:24 UTC
+++ misc//build/boost_1_64_0/boost/serialization/array.hpp
@@ -23,6 +23,8 @@ namespace std{ 
 } // namespace std
 #endif
 
+#include <boost/serialization/array_wrapper.hpp>
+
 #ifndef BOOST_NO_CXX11_HDR_ARRAY
 
 #include <array>
