diff -ur misc/nss-3.39/nss/lib/sqlite/config.mk misc/build/nss-3.39/nss/lib/sqlite/config.mk
--- misc/nss-3.39/nss/lib/sqlite/config.mk	2018-08-31 05:55:53.000000000 -0700
+++ misc/build/nss-3.39/nss/lib/sqlite/config.mk	2018-09-10 10:36:01.074375000 -0700
@@ -35,9 +35,9 @@
 # OSAtomicCompareAndSwapPtrBarrier, which is only available on Mac OS X 10.5
 # (Darwin 9.0) and later. Define SQLITE_WITHOUT_ZONEMALLOC to disable
 # that code for older versions of Mac OS X. See bug 820374.
-DARWIN_VER_MAJOR := $(shell uname -r | cut -f1 -d.)
-DARWIN_LT_9 := $(shell [ $(DARWIN_VER_MAJOR) -lt 9 ] && echo true)
-ifeq ($(DARWIN_LT_9),true)
+OSX_TARGET_MINOR := $(shell echo $(MACOSX_DEPLOYMENT_TARGET) | cut -f2 -d.)
+OSX_MINOR_LT_5 := $(shell [ $(OSX_TARGET_MINOR) -lt 5 ] && echo true)
+ifeq ($(OSX_MINOR_LT_5),true)
 OS_CFLAGS += -DSQLITE_WITHOUT_ZONEMALLOC
 endif
 endif # Darwin
--- misc/nss-3.39/nss/coreconf/UNIX.mk	2021-09-07 01:32:04.000000000 -0400
+++ misc/build/nss-3.39/nss/coreconf/UNIX.mk	2021-09-07 01:32:11.000000000 -0400
@@ -63,4 +63,4 @@
 if test ! -d $(@D); then rm -rf $(@D); $(NSINSTALL) -D $(@D); fi
 endef
 
-include $(CORE_DEPTH)/coreconf/Werror.mk
+#include $(CORE_DEPTH)/coreconf/Werror.mk
--- misc/nss-3.39/nss/coreconf/Darwin.mk	2021-09-07 14:24:22.000000000 -0400
+++ misc/build/nss-3.39/nss/coreconf/Darwin.mk	2021-09-07 14:24:30.000000000 -0400
@@ -8,6 +8,9 @@
 RANLIB ?= ranlib
 
 include $(CORE_DEPTH)/coreconf/UNIX.mk
-include $(CORE_DEPTH)/coreconf/Werror.mk
+#include $(CORE_DEPTH)/coreconf/Werror.mk
+
+CXXFLAGS += -stdlib=libc++
+LDFLAGS += -stdlib=libc++
 
 DEFAULT_COMPILER = gcc
@@ -108,7 +111,7 @@
 # May override this with different compatibility and current version numbers.
 DARWIN_DYLIB_VERSIONS = -compatibility_version 1 -current_version 1
 # May override this with -bundle to create a loadable module.
-DSO_LDOPTS	= -dynamiclib $(DARWIN_DYLIB_VERSIONS) -install_name @executable_path/$(notdir $@) -headerpad_max_install_names
+DSO_LDOPTS	= -dynamiclib $(DARWIN_DYLIB_VERSIONS) -install_name @executable_path/$(notdir $@) -headerpad_max_install_names -stdlib=libc++
 
 ifdef USE_GCOV
    OS_CFLAGS += --coverage
