--- misc/CoinMP-1.7.6/CoinUtils/src/CoinSignal.hpp	2011-01-03 20:31:00.000000000 -0300
+++ misc/build/CoinMP-1.7.6/CoinUtils/src/CoinSignal.hpp	2014-02-01 02:02:10.596696644 -0300
@@ -95,7 +95,10 @@
 
 //-----------------------------------------------------------------------------
 
-#if defined(__MACH__) && defined(__GNUC__)
+#if defined(__clang__)
+   typedef void(*CoinSighandler_t)(int);
+#  define CoinSighandler_t_defined
+#elif defined(__MACH__) && defined(__GNUC__)
    typedef typeof(SIG_DFL) CoinSighandler_t;
 #  define CoinSighandler_t_defined
 #endif
