diff -ur misc/libxslt-1.1.34/win32/configure.js misc/build/libxslt-1.1.34/win32/configure.js
--- misc/libxslt-1.1.32/win34/configure.js	2017-10-26 00:55:47.000000000 -0700
+++ misc/build/libxslt-1.1.34/win32/configure.js	2019-11-08 16:16:13.055825865 -0800
@@ -51,7 +51,7 @@
 var dirSep = "\\";
 var compiler = "msvc";
 var cruntime = "/MD";
-var vcmanifest = false;
+var vcmanifest = true;
 var buildDebug = 0;
 var buildStatic = 0;
 var buildPrefix = ".";
@@ -240,6 +240,8 @@
 			of.WriteLine(s.replace(/\@WITH_DEBUGGER\@/, withDebugger? "1" : "0"));
 		} else if (s.search(/\@WITH_MODULES\@/) != -1) {
 			of.WriteLine(s.replace(/\@WITH_MODULES\@/, withModules? "1" : "0"));
+		} else if (s.search(/\@WITH_PROFILER\@/) != -1) {
+			of.WriteLine(s.replace(/\@WITH_PROFILER\@/, "0"));
 		} else if (s.search(/\@LIBXSLT_DEFAULT_PLUGINS_PATH\@/) != -1) {
 			of.WriteLine(s.replace(/\@LIBXSLT_DEFAULT_PLUGINS_PATH\@/, "NULL"));
 		} else
