diff -ur misc/libxml2-2.9.10/include/libxml/xmlversion.h misc/build/libxml2-2.9.10/include/libxml/xmlversion.h
--- misc/libxml2-2.9.10/include/libxml/xmlversion.h	2018-03-05 07:54:29.000000000 -0800
+++ misc/build/libxml2-2.9.10/include/libxml/xmlversion.h	2018-08-22 22:53:15.488158000 -0700
@@ -270,7 +270,7 @@
  *
  * Whether iconv support is available
  */
-#if 1
+#if 0
 #define LIBXML_ICONV_ENABLED
 #endif
 
@@ -297,7 +297,7 @@
  *
  * Whether Debugging module is configured in
  */
-#if 1
+#if 0
 #define LIBXML_DEBUG_ENABLED
 #endif
 
diff -ur misc/libxml2-2.9.10/xml2-config.in misc/build/libxml2-2.9.10/xml2-config.in
--- misc/libxml2-2.9.10/xml2-config.in	2016-06-07 03:04:14.000000000 -0700
+++ misc/build/libxml2-2.9.10/xml2-config.in	2019-11-07 22:50:00.185459398 -0800
@@ -1,9 +1,14 @@
 #! /bin/sh
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-includedir=@includedir@
-libdir=@libdir@
+#prefix=@prefix@
+#exec_prefix=@exec_prefix@
+#includedir=@includedir@
+#libdir=@libdir@
+
+prefix=${SOLARVERSION}/${INPATH}
+exec_prefix=${SOLARVERSION}/${INPATH}
+includedir=${SOLARVERSION}/${INPATH}/inc${UPDMINOREXT}/external
+libdir=${SOLARVERSION}/${INPATH}/lib${UPDMINOREXT}
 
 usage()
 {
@@ -65,7 +70,8 @@
 	;;
 
     --cflags)
-        cflags="@XML_INCLUDEDIR@ @XML_CFLAGS@"
+        cflags="-I${includedir}"
+#        cflags="@XML_INCLUDEDIR@ @XML_CFLAGS@"
        	;;
 
     --libtool-libs)
@@ -80,18 +86,19 @@
        	;;
 
     --libs)
-        if [ "$2" = "--dynamic" ]; then
-            shift
-            libs="@XML_LIBS@"
-        else
-            libs="@XML_LIBS@ @XML_PRIVATE_LIBS@ @MODULE_PLATFORM_LIBS@"
-        fi
-
-        if [ "@XML_LIBDIR@" != "-L/usr/lib" -a "@XML_LIBDIR@" != "-L/usr/lib64" ]; then
-            libs="@XML_LIBDIR@ $libs"
-        fi
-
-        libs="$libs @WIN32_EXTRA_LIBADD@"
+        libs="-L${libdir} ${LIBXML2LIB} -lm"
+        if [ "$2" = "--dynamic" ]; then
+            shift
+#            libs="@XML_LIBS@"
+#        else
+#            libs="@XML_LIBS@ @XML_PRIVATE_LIBS@ @MODULE_PLATFORM_LIBS@"
+        fi
+#
+#        if [ "@XML_LIBDIR@" != "-L/usr/lib" -a "@XML_LIBDIR@" != "-L/usr/lib64" ]; then
+#            libs="@XML_LIBDIR@ $libs"
+#        fi
+#
+#        libs="$libs @WIN32_EXTRA_LIBADD@"
         ;;
 
     *)
