--- misc/serf-1.2.1/serf_private.h	2013-06-03 14:22:51.000000000 -0300
+++ misc/build/serf-1.2.1/serf_private.h	2013-06-07 12:33:34.432688365 -0300
@@ -32,12 +32,18 @@
 
 /* Internal logging facilities, set flag to 1 to enable console logging for
    the selected component. */
-#define SSL_VERBOSE 0
-#define SSL_MSG_VERBOSE 0  /* logs decrypted requests and responses. */
-#define SOCK_VERBOSE 0
-#define SOCK_MSG_VERBOSE 0 /* logs bytes received from or written to a socket. */
-#define CONN_VERBOSE 0
-#define AUTH_VERBOSE 0
+#ifdef ENABLE_SERF_VERBOSE
+#define SERF_VERBOSE_FLAG 1
+#else
+#define SERF_VERBOSE_FLAG 0
+#endif
+
+#define SSL_VERBOSE         SERF_VERBOSE_FLAG
+#define SSL_MSG_VERBOSE     SERF_VERBOSE_FLAG  /* logs decrypted requests and responses. */
+#define SOCK_VERBOSE        SERF_VERBOSE_FLAG
+#define SOCK_MSG_VERBOSE    SERF_VERBOSE_FLAG /* logs bytes received from or written to a socket. */
+#define CONN_VERBOSE        SERF_VERBOSE_FLAG
+#define AUTH_VERBOSE        SERF_VERBOSE_FLAG
 
 
 typedef struct serf__authn_scheme_t serf__authn_scheme_t;
