--- misc/serf-1.2.0/configure.in	2012-12-08 16:45:28.000000000 -0300
+++ misc/build/serf-1.2.0/configure.in	2013-02-25 11:11:04.106764823 -0300
@@ -199,7 +199,7 @@ AC_SEARCH_LIBS(fabs, m)
 
 libs=""
 if test -n "$apu_config"; then
-  APR_ADDTO(libs, [`$apu_config --link-libtool --libs`])
+  APR_ADDTO(libs, [`$apu_config --link-libtool`])
 fi
 APR_ADDTO(libs, [`$apr_config --link-libtool --libs` $LIBS])
 
