--- misc/hyphen-2.8.8/makefile.mk	2010-12-02 10:35:40.265625000 +0100
+++ misc/build/hyphen-2.8.8/makefile.mk	2010-12-02 10:25:45.750000000 +0100
@@ -1 +1,28 @@
-dummy
+PRJ = ..$/..$/..$/..
+
+PRJNAME = hyphen
+TARGET  = hyphen
+CFLAGSCALL=gsd
+
+USE_DEFFILE=TRUE
+EXTERNAL_WARNINGS_NOT_ERRORS := TRUE
+UWINAPILIB=
+
+.INCLUDE : settings.mk
+
+# --- Files --------------------------------------------------------
+
+# !! not to be compiled because those belong to a stand alone programs: !!
+#        $(SLO)$/createfp.obj\
+#        $(SLO)$/testtextcat.obj
+
+SLOFILES=   \
+        $(SLO)$/hyphen.obj\
+        $(SLO)$/hnjalloc.obj
+
+# --- Targets ------------------------------------------------------
+
+ALL: ALLTAR
+
+.INCLUDE : target.mk
+
