diff -ru misc/Python-3.11.15/Modules/Setup misc/build/Python-3.11.15/Modules/Setup
--- misc/Python-3.11.15/Modules/Setup	2025-01-01 00:00:00.000000000 +0000
+++ misc/build/Python-3.11.15/Modules/Setup	2025-01-01 00:00:00.000000000 +0000
@@ -163,11 +163,11 @@

 # hashing builtins
 #_blake2 _blake2/blake2module.c _blake2/blake2b_impl.c _blake2/blake2s_impl.c
-#_md5 md5module.c
-#_sha1 sha1module.c
-#_sha256 sha256module.c
-#_sha512 sha512module.c
-#_sha3 _sha3/sha3module.c
+_md5 md5module.c
+_sha1 sha1module.c
+_sha256 sha256module.c
+_sha512 sha512module.c
+_sha3 _sha3/sha3module.c

 # text encodings and unicode
 #_codecs_cn cjkcodecs/_codecs_cn.c
